src/app/static/src/app/store/genericChart/mutations.ts

```diff
@@ -21,8 +21,8 @@ export const mutations: MutationTree<GenericChartState> = {
     [GenericChartMutation.SetDataset](state: GenericChartState, action: PayloadWithType<SetDatasetPayload>) {
         state.datasets[action.payload.datasetId] = action.payload.dataset;
     },
-    [GenericChartMutation.ClearDataset](state: GenericChartState) {
-        state.datasets = {};
+    [GenericChartMutation.ClearDataset](state: GenericChartState, action: PayloadWithType<string>) {
+        state.datasets[action.payload] = {} as GenericChartDataset;
```

Comment:
It would be better to remove this from the dictionary entirely rather than replacing it with {}, which will fail tests for falsiness - this will restore it to the state it was in before the dataset was fetched. 
```suggestion
        delete state.datasets[action.payload];
```